src/webgpu/api/validation/capability_checks/limits/maxStorageBufferBindingSize.spec.ts

```diff
@@ -0,0 +1,166 @@
+import { keysOf } from '../../../../../common/util/data_tables.js';
+import { roundDown } from '../../../../util/math.js';
+
+import {
+  kLimitBaseParams,
+  makeLimitTestGroup,
+  LimitValueTest,
+  TestValue,
+  kLimitValueTestKeys,
+} from './limit_utils.js';
+
+const BufferParts = {
+  wholeBuffer: true,
+  biggerBufferWithOffset: true,
+};
+type BufferPart = keyof typeof BufferParts;
+const kBufferPartsKeys = keysOf(BufferParts);
+
+function getSizeAndOffsetForBufferPart(device: GPUDevice, bufferPart: BufferPart, size: number) {
+  const align = device.limits.minUniformBufferOffsetAlignment;
+  switch (bufferPart) {
+    case 'wholeBuffer':
+      return { size, offset: 0 };
+    case 'biggerBufferWithOffset':
+      return { size: size + align, offset: align };
+  }
+}
+
+const kStorageBufferRequiredSizeAlignment = 4;
+
+function getDeviceLimitToRequest(
+  limitValueTest: LimitValueTest,
+  defaultLimit: number,
+  maximumLimit: number
+) {
+  switch (limitValueTest) {
+    case 'atDefault':
+      return defaultLimit;
+    case 'underDefault':
+      return defaultLimit - kStorageBufferRequiredSizeAlignment;
+    case 'betweenDefaultAndMaximum':
+      return Math.floor((defaultLimit + maximumLimit) / 2);
+    case 'atMaximum':
+      return maximumLimit;
+    case 'overMaximum':
+      return maximumLimit + kStorageBufferRequiredSizeAlignment;
+  }
+}
+
+function getTestValue(testValueName: TestValue, requestedLimit: number) {
+  switch (testValueName) {
+    case 'atLimit':
+      return roundDown(requestedLimit, kStorageBufferRequiredSizeAlignment);
+    case 'overLimit':
+      // Note: the requestedLimit might not meet alignment requirements.
+      return roundDown(
+        requestedLimit + kStorageBufferRequiredSizeAlignment,
+        kStorageBufferRequiredSizeAlignment
+      );
+  }
+}
+
+function getDeviceLimitToRequestAndValueToTest(
+  limitValueTest: LimitValueTest,
+  testValueName: TestValue,
+  defaultLimit: number,
+  maximumLimit: number
+) {
+  const requestedLimit = getDeviceLimitToRequest(limitValueTest, defaultLimit, maximumLimit);
+  return {
+    requestedLimit,
+    testValue: getTestValue(testValueName, requestedLimit),
+  };
+}
+
+const limit = 'maxStorageBufferBindingSize';
+export const { g, description } = makeLimitTestGroup(limit);
+
+g.test('createBindGroup,at_over')
```

Comment:
I think 'at_over' is redundant because the test params will have 'atLimit' and 'overLimit' already. Though looking at all the other tests in the folder I see that this is already the naming convention... So I guess it's fine to keep it.

Reply:
same as above, @kainino0x suggested [in another PR](https://github.com/gpuweb/cts/pull/2227#discussion_r1099481413) this naming convention.